### PR TITLE
Cache all needed projects for navbar layout

### PR DESCRIPTION
**Pull request by cproensa**
Reduce DB query count by caching all needed projects for the navbar layout.

Fixes: #23237

After:
seleccion_231

Before:
seleccion_230

### CODE REVIEW

**Comment by cproensa** [comment]
Added a second commit, where view_all_bug_page may read the project rows as part of the filter query, before the layout for navbar is done.

File: core/filter_api.php [hunk 1]
```diff
@@ -1492,6 +1492,7 @@ function filter_get_bug_rows_query_clauses( array $p_filter, $p_project_id = nul
 		$t_private_and_public_project_ids = array();
 		$t_limited_projects = array();
 
+		project_cache_array_rows( $t_project_ids );
```
Comment (vboctor) [inline core/filter_api.php]: It would be good to add the comments about why this is called, similar to the comment in the commit message.

Comment (cproensa) [inline core/filter_api.php]: done

**Comment by cproensa** [comment]
@atrol seems like this PR has been merged, but not closed.

**Pull request closed without merging**